### PR TITLE
More accurate measuretext

### PR DESCRIPTION
**Pull request by chearon**
The values from `measureText()` are using Pango methods which round to the nearest pixel, since their use case is actual device pixels. This makes it like the browser and returns fractional pixels, which enables me to make text wrapping that matches the browser identically.

First commit is just minor cleanup

PS the measureText() tests for the additional TextMetrics properties are still failing on macOS

### CODE REVIEW

File: src/CanvasRenderingContext2d.cc [hunk 1]
```diff
@@ -2073,14 +2073,29 @@ NAN_METHOD(Context2d::MeasureText) {
...
+
+  logical_rect.x = 1.0 * _logical_rect.x / PANGO_SCALE;
+  logical_rect.y = 1.0 * _logical_rect.y / PANGO_SCALE;
+  logical_rect.width = 1.0 * _logical_rect.width / PANGO_SCALE;
+  logical_rect.height = 1.0 * _logical_rect.height / PANGO_SCALE;
+
+  ink_rect.x = 1.0 * _ink_rect.x / PANGO_SCALE;
+  ink_rect.y = 1.0 * _ink_rect.y / PANGO_SCALE;
+  ink_rect.width = 1.0 * _ink_rect.width / PANGO_SCALE;
+  ink_rect.height = 1.0 * _ink_rect.height / PANGO_SCALE;
```
Comment (zbjornson) [inline src/CanvasRenderingContext2d.cc]: Think it's preferable to explicitly cast one or both operands to a float/double. You could cast `PANGO_SCALE` once and be done.

Comment (chearon) [reply src/CanvasRenderingContext2d.cc]: Makes sense, updated PR.

**Comment by chearon** [comment]
Before this PR (`measureText('Hello!')`):

```
{ width: 26,
  actualBoundingBoxLeft: 0,
  actualBoundingBoxRight: 26,
  actualBoundingBoxAscent: 7,
  actualBoundingBoxDescent: 1,
  emHeightAscent: 7,
  emHeightDescent: 3,
  alphabeticBaseline: -0 }
```

After:

```
{ width: 25.56640625,
  actualBoundingBoxLeft: 0,
  actualBoundingBoxRight: 25.56640625,
  actualBoundingBoxAscent: 6.47265625,
  actualBoundingBoxDescent: 0.890625,
  emHeightAscent: 7,
  emHeightDescent: 3,
  alphabeticBaseline: -0.7001953125 }
```

**Comment by LinusU** [comment]
Cheers!

**Pull request merged**